src/use-swr.ts

```diff
@@ -473,6 +473,16 @@ function useSWR<Data = any, Error = any>(
       timeout = setTimeout(tick, config.refreshInterval)
     }
 
+    // set up reconnecting when the browser regains network connection
+    let reconnect = null
+    if (config.revalidateOnReconnect) {
+      const reconnectEvent = async () => {
+        await softRevalidate()
+      }
+
+      reconnect = addEventListener('online', reconnectEvent)
```

Comment:
```suggestion
      reconnect = addEventListener('online', softRevalidate)
```

And we can remove `reconnectEvent`.

Reply:
All other things looks good! Thank you @daneden ❤️

Reply:
Ah nice, I wondered if it was necessary to wrap it in an async function. Will update now!

Reply:
Ah nice! I wondered if I had to wrap this in an async function. Just pushed this change.